### PR TITLE
neo_simulation2 Humble branch updated with arm support and urdf files updated to modular xacros

### LINKED ISSUE #85: refactor the urdf to xacro code for mpo robots and add logic to include arm
Change the urdf to xacro
Split into separate files
Make code modular and efficient 
Add logic to include UR and Elite arms to the xacro

### LINKED ISSUE #98: add IMU and Laserscanner xacro
Make it available as options to the users, depending upon their needs they can select. It should be a launch argument, which would then be explicitly converted into a xacro argument, and passed on to the urdf.xacro of the robot,

- [x] S300
- [ ] IMU

Create individual xacros for laserscanners and IMU. Can be as flexible as it can.

### LINKED ISSUE #99: update caster model for MPO-700
use the same one as that of the real robot..

### PR DESCRIPTION
**Pull request by AdarshKaran**
arm type validation for mpo_500 and mpo_700, ROS2 control node integration for UR arms and modularization of robot description files into xacro format to reduce redundancy. Additionally, the package structure has been reorganized to support Xacro files, and the caster model for MPO-700 has been updated (see issue #99).

### CODE REVIEW

**Comment by padhupradheep** [comment]
https://github.com/user-attachments/assets/470d651c-3f92-4663-aba3-fca847d14be6

here is a sample video

File: configs/ur_config/ur10/moveit_controllers.yaml
Comment (padhupradheep) [inline configs/ur_config/ur10/moveit_controllers.yaml:1]: Please remove this file, it is not required here.

File: configs/ur_config/ompl_planning.yaml
Comment (padhupradheep) [inline configs/ur_config/ompl_planning.yaml:1]: This file is also not required, please remove it as well.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,60 @@
 This simulation package provides a quick and easy way to try out the autonomous mobile robots from Neobotix in ROS-2. It comes with the most commonly used configuration but is open for any kind of modification.
 
 Please find our documentations in https://neobotix-docs.de/ros/ros2/simulation.html
+
+# Neobotix Neo Simulation2 Package - Contribution Overview
```
Comment (padhupradheep) [inline README.md]: Could you please check our updated documentation website and tell me what exact changes that needs to be made ? 

We don't want an additional readme.

**Review by padhupradheep — CHANGES_REQUESTED** [review]
Overall LGTM. Just some minor changes before this goes in.

File: configs/ur_config/ur10/moveit_controllers.yaml
Comment (AdarshKaran) [reply configs/ur_config/ur10/moveit_controllers.yaml:1]: i have removed both the moveit files Pradheep.

**Comment by padhupradheep** [comment]
I have tested the changes @AdarshKaran, works really good. This is how I wanted it to be exact. One minor point, if a user uses the `arm_type` argument with an MP-500 or MP-400. I want to leave it as such, we can fix that later when someone complains.

**Pull request merged**